pkg/operator/controller/nodeobservabilityrun/controller.go

```diff
@@ -115,29 +126,34 @@ func (r *NodeObservabilityRunReconciler) Reconcile(ctx context.Context, req ctrl
 		r.Log.V(3).Info("Run for this instance has been completed already")
 		// check the CR if the RestoreMCOStateAfterRun flag is set
 		if instance.Spec.RestoreMCOStateAfterRun {
-			mco := r.desiredMCO()
-			if err := r.Delete(ctx, mco); err != nil {
-				return ctrl.Result{}, fmt.Errorf("failed to remove profiling MCO CR %s : %w", instance.Name, err)
+			err := r.disableCrioKubeletProfile(ctx)
+			if err != nil {
+				return ctrl.Result{}, err
 			}
-			mcp := r.MCOReconciler.GetProfilingMCP(machineconfigcontroller.ProfilingMCPName)
-			err = r.MCOReconciler.DeleteMCP(ctx, mcp)
+
+			disabled, err := r.checkNOMCStatus(ctx, nodeobservabilityv1alpha1.DebugDisabled)
 			if err != nil {
-				r.Log.Error(err, "deleting machineconfigpool")
+				r.Log.Error(err, "NodeObservabilityMachineConfig disable status check failed")
+				return ctrl.Result{}, err
+			}
+			if disabled {
+				r.Log.Info("NodeObservabilityMachineConfig CrioKubeletProfile disabled")
+				nomc := r.desiredMCO()
+				if err := r.deleteNOMC(ctx, nomc); err != nil {
+					if !errors.IsNotFound(err) {
+						return ctrl.Result{}, err
+					}
+				}
+
+				instance.Spec.RunType = ""
```

Comment:
This looks like a noop

Reply:
This needs to be discussed, shouldn't have included in PR. But thought to discuss and update it. Need a way to disable it, as during reconcile after 30minutes and removing all configs, if set everything will be repeated.